mining/src/cache.rs

```diff
@@ -0,0 +1,45 @@
+use consensus_core::block::BlockTemplate;
+use std::{
+    rc::Rc,
+    time::{SystemTime, UNIX_EPOCH},
+};
+
+/// CACHE_LIFETIME indicates the default duration in milliseconds after which the cached data expires.
+const DEFAULT_CACHE_LIFETIME: u64 = 1_000;
+
+pub(crate) struct BlockTemplateCache {
+    /// Time, in milliseconds, when the cache was last updated
+    last_update_time: u64,
+    block_template: Option<Rc<BlockTemplate>>,
```

Comment:
When `MiningManager` is used and passed between threads this line will not compile bcs of the Rc. It must be an Arc